tests/functional/behave_features/common/utils/owner_file_submission.py

```diff
@@ -0,0 +1,351 @@
+        with SetDirectory(Path(self.temp_dir.name)):
+            # Create the OWNERS file from the string template
+            values = {
+                "bot_name": self.secrets.bot_name,
+                "vendor": vendor_label,
+                "vendor_pretty": vendor_name,
+                "chart_name": chart_name,
+                "public_key": "null",
+                "provider_delivery": "false",
+            }
+            content = Template(base_owners_file).substitute(values)
+            logging.debug(f"OWNERS File Content:\n{content}")
+            pathlib.Path(chart_directory).mkdir(parents=True, exist_ok=True)
+            with open(f"{chart_directory}/OWNERS", "w+") as fd:
+                fd.write(content)
+
+            logging.info(f"Push OWNERS file to '{self.secrets.test_repo}:{pr_branch}'")
+            worktree_repo = git.Repo()
+            worktree_repo.git.add(f"{chart_directory}/OWNERS")
+            worktree_repo.git.commit("-m", f"Add {chart_name} OWNERS file")
+            self.repo_manager.push_branch(
+                self.secrets.test_repo, pr_branch, repo=worktree_repo
+            )
+
+    def send_pull_request(self):
+        """Sends a pull request to be evaluated for CI."""
+        self.pull_request = self._send_pull_request(
+            self.secrets.test_repo,
+            self.secrets.base_branch,
+            self.secrets.pr_branch,
+            self.secrets.bot_token,
+            os.environ.get("PR_BODY"),
+        )
+        return self.pull_request
+
+    def _send_pull_request(
+        self,
+        remote_repo: str,
+        base_branch: str,
+        pr_branch: str,
+        gh_token: str,
+        pr_body: str,
+    ):
+        """Sends a pull request using the GitHub API.
+
+        The branches should already exist in the remotes. This only sends the
+        pull request API call referring to two pre-existing branches in
+        remote_repo.
+
+        Pull requests created with this method are not tracked for cleanup.
+        When related branches are deleted, GitHub will automatically close the PR.
+
+        Args:
+            remote_repo: the org/repo string where the PR should be pushed.
+            base_branch: the branch receiving the PR.
+            pr_branch: the modified content to send to base_branch.
+            pr_body: the text to send in the PR body.
+
+        Raises:
+            AssertionError in cases where the pull requests could not be sent.
+
+        Returns:
+            The PR number for the generated PR.
+        """
+        data = {
+            "head": pr_branch,
+            "base": base_branch,
+            "title": base_branch,
+            "body": pr_body,
+        }
+        logging.debug(f"Pull Request Body Text: {pr_body}")
+        logging.info(
+            f"Create PR from '{remote_repo}:{pr_branch}' to '{remote_repo}:{base_branch}'"
+        )
+        r = github.github_api("post", f"repos/{remote_repo}/pulls", gh_token, json=data)
```

Comment:
:arrow_up: trying to make this comment more visible in the cluttered GitHub UI :)

Reply:
Will do, my apologies

Reply:
This has been addressed. Note that I do catch a generic exception here, but that's because the requests exception would be the only reason `requests` would be imported in this library (requests.exception.RequestException).

At some point, if the github library emits its own exceptions, we can switch these out to call that.